include/libp2p/peer/stream_protocols.hpp

```diff
@@ -16,5 +16,11 @@ namespace libp2p {
    * Used by `Host::newStream`, `Dialer::newStream`, `Host::setProtocolHandler`,
    * `Router::setProtocolHandler`.
    */
-  using StreamProtocols = std::vector<peer::ProtocolName>;
+  struct StreamProtocols {
```

Comment:
Revert.
`StreamProtocols` is just data type (like `vector`),
it's ambigous to inject it.

Create `struct Config` for identify (see ping or kademlia for example)

Reply:
Might to use Tagged wrapping-class from kagome